docs: Add example to tutorial for interfaces using closures (#25068)
nim-lang/Nim: doc/tut2.md

(*is-a* relation) for simple code reuse. Since objects are value types in
Nim, composition is as efficient as inheritance.

Interfaces
----------

> Closely related concepts of abstract classes, protocols, traits, and interfaces are

Reviewer: ```suggestion
Concepts like abstract classes, protocols, traits, and interfaces can be
```

> simulated in nim as objects of closures:

Reviewer: ```suggestion
simulated as objects of closures:
```
